pkg/metallb/frrnodestate.go

```diff
@@ -0,0 +1,129 @@
+package metallb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/schemes/metallb/frrtypes"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// FrrNodeStateBuilder provides struct for FrrNodeState object which contains connection to cluster and
+// frrconfiguration definitions.
+type FrrNodeStateBuilder struct {
+	Definition *frrtypes.FRRNodeState
+	Object     *frrtypes.FRRNodeState
+	apiClient  runtimeClient.Client
+	errorMsg   string
+}
+
+// PullFrrNodeState retrieves an existing FrrNodeState object from the cluster.
+func PullFrrNodeState(apiClient *clients.Settings, name string) (*FrrNodeStateBuilder, error) {
+	glog.V(100).Infof("Pulling FrrNodeState object name:%s", name)
+
+	if apiClient == nil {
+		glog.V(100).Infof("The apiClient cannot be nil")
+
+		return nil, fmt.Errorf("the apiClient cannot be nil")
+	}
+
+	err := apiClient.AttachScheme(frrtypes.AddToScheme)
+	if err != nil {
+		glog.V(100).Infof("Failed to add nmstate v1 scheme to client schemes")
+
+		return nil, err
+	}
+
+	frrStateBuilder := FrrNodeStateBuilder{
+		apiClient: apiClient.Client,
+		Definition: &frrtypes.FRRNodeState{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: metav1.NamespaceNone,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the FrrNodeState is empty")
+
+		return nil, fmt.Errorf("frrNodeState 'name' cannot be empty")
+	}
+
+	if !frrStateBuilder.Exists() {
+		return nil, fmt.Errorf("frrNodeState object %s does not exist", name)
+	}
+
```

Comment:
Make sure to set the definition to the object after checking it exists so that the definition is updated
```go
builder.Definition = builder.Object
```

Reply:
added